### PR TITLE
Fix type of some RNA fusion variant keys

### LINKED ISSUE #4281: Tool hits field for a fusion variant should be an integer, not a string
Example from a demo variant

image

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Converts RNA fusions variants `tool_hits, fusion_score` keys from string to numbers - Fix #4281 - 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
Main branch:

image


image


This branch:

image

image

File: scout/parse/variant/variant.py [hunk 1]
```diff
@@ -458,10 +458,10 @@ def set_found_db(found_db_info: Optional[str]) -> Optional[List[str]]:
         str, replace_nan(variant.INFO.get("TRANSCRIPT_ID_B", ""))
     )
     parsed_variant["exon_number_a"] = call_safe(
-        str, call_safe(int, replace_nan(variant.INFO.get("EXON_NUMBER_A", "")))
+        int, call_safe(int, replace_nan(variant.INFO.get("EXON_NUMBER_A", "")))
```
Comment (fevac) [inline scout/parse/variant/variant.py:461]: these are numbers but they get store into the exon field of transcripts which is a string

**Review by fevac — APPROVED** [review]
Nice cleanup! ⭐ My only comment is about the exon field. See below

File: scout/parse/variant/variant.py (hunk 1 above)
Comment (northwestwitch) [reply scout/parse/variant/variant.py:461]: But in the scout document should be numbers again, no?

Comment (northwestwitch) [reply scout/parse/variant/variant.py]: Checked other variant types, like SVs and exons are strings in there (i.e. "2/7"). So I've reverted these values as they were before

**Pull request merged**